File: backend/runnity/src/main/java/com/runnity/broadcast/dto/BroadcastJoinResponse.java
package com.runnity.broadcast.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "중계방 입장 응답 dto")
public class BroadcastJoinResponse {

    @Schema(
            description = "WebSocket 접속 URL",
            example = "ws://43...:8080/ws"
    )
    private String wsUrl;

    @Schema(
            description = "구독해야 할 STOMP topic",
            example = "/topic/broadcast/13"
    )
    private String topic;

    @Schema(
            description = "입장한 챌린지 ID",
            example = "13"
    )
    private Long challengeId;
}

File: frontend/app/src/main/java/com/example/runnity/data/model/response/BroadcastJoinResponse.kt
package com.example.runnity.data.model.response

import com.google.gson.annotations.SerializedName

/**
 * 중계방 입장 API 응답 모델
 * GET /api/v1/broadcast/join
 */
data class BroadcastJoinResponse(
    val wsUrl: String,
    val topic: String,
    val challengeId: Long,
    val hlsUrl: String?,      // 백엔드에서 추가
    val title: String?,       // 선택
    val distance: String?,    // 선택 (예: "5km")
    val totalDistanceMeter: Int?
)
